packages/config-resolver/src/EndpointsConfig.ts

```diff
@@ -13,9 +13,9 @@ export interface EndpointsInputConfig {
 }
 
 interface PreviouslyResolved {
-  regionInfoProvider: RegionInfoProvider;
+  regionInfoProvider?: RegionInfoProvider;
   urlParser: UrlParser;
-  region: Provider<string>;
+  region?: Provider<string>;
 }
```

Comment:
For the JS folks - will this cause any issues?

Reply:
See my refactoring in https://github.com/aws/aws-sdk-js-v3/pull/2287 to remove this concern.